src/claimtriedb.cpp

```diff
@@ -0,0 +1,215 @@
+
+#include "claimtriedb.h"
+#include "claimtrie.h"
+
+#include <iterator>
+#include <boost/scoped_ptr.hpp>
+#include <list>
+
+/**
+ * function to provide unique char of 2 types
+ */
+template <class T1, class T2>
+unsigned char hashType();
+
+/// Concrete implementation of buffer
+template <typename K, typename V>
+struct CActualBuffer : public CAbstractBuffer
```

Comment:
What if this inherited from std::map instead? You wouldn't need the find and insert methods below. You wouldn't need the dataType stuff. It seems like that would be less code.

Reply:
Map is ordered i.e. it'll reorder every insert, list benefits speed over it O(log N) vs O(N)

Reply:
Your current use of a list is O(1) insertion time; it's the lookup time that suffers. Once we get the upstream c++11, we could use an unordered_list, which is definitely faster. However, the number of buffers we have is small; none of these solutions are going to be a performance problem. Hence, I prefer the "less code" option.